nixos/modules/services/misc/ananicy.nix

```
@@ -0,0 +1,107 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.ananicy;
+  configFile = pkgs.writeText "ananicy.conf" (generators.toKeyValue { } cfg.settings);
+  extraRules = pkgs.writeText "extraRules" cfg.extraRules;
+  servicename = if ((lib.getName cfg.package) == (lib.getName pkgs.ananicy-cpp)) then "ananicy-cpp" else "ananicy";
+in
+{
+  options = {
+    services.ananicy = {
+      enable = mkEnableOption "Ananicy, a auto nice daemon";
+
+      package = mkOption {
+        type = types.package;
+        default = pkgs.ananicy;
+        defaultText = literalExpression "pkgs.ananicy";
+        example = literalExpression "pkgs.ananicy-cpp";
+        description = ''
+          Which ananicy package to use.
+        '';
+      };
```

Since the idea of this setting is just to select between the `ananicy` variations, maybe it would be better to have instead:

```suggestion
      useAnanicyCpp = mkOption {
        type = types.boolean;
        default = false;
        description = ''
          Use ananicy-cpp instead of ananicy.
        '';
      };
```

---

We could still have `package` option, but use them as they're used in other modules (to allow the user to overwrite the package inside a module without needing to use other methods like overlays).

---

i would prefer to avoid adding additional options as the `package` already does what we want